sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/implementation/RedisCacheImpl.java

```diff
@@ -568,23 +568,27 @@ public RedisCacheImpl update() {
 
     @Override
     public Mono<RedisCache> updateResourceAsync() {
-        final RedisCacheImpl self = this;
+        updateParameters.withTags(this.inner().tags());
         return this.manager().inner().getRedis().updateAsync(resourceGroupName(), name(), updateParameters)
                 .map(innerToFluentMap(this))
-                .doOnNext(redisCache -> {
-                    while (!redisCache.provisioningState().equalsIgnoreCase("Succeeded")) {
-                        SdkContext.sleep(30 * 1000);
-
-                        RedisResourceInner innerResource = self.manager().inner().getRedis().getByResourceGroup(resourceGroupName(), name());
-                        ((RedisCacheImpl) redisCache).setInner(innerResource);
-                        self.setInner(innerResource);
-                        self.patchScheduleAdded = false;
+                .filter(redisCache -> !redisCache.provisioningState().equalsIgnoreCase(ProvisioningState.SUCCEEDED.toString()))
+                .flatMap(redisCache -> {
+                    SdkContext.sleep(30 * 1000);
+                    this.patchScheduleAdded = false;
+                    return this.manager().inner().getRedis().getByResourceGroupAsync(resourceGroupName(), name())
```

Comment:
Another fake LRO?

Reply:
Yes, the redis doesn't go through official LRO, like cosmos